G1ANT.Addon.Net/SimplifiedMessageSummary.cs

```diff
@@ -1,27 +1,321 @@
         public IMailFolder Folder;
-        public ListStructure MessageAttachments;
 
-        public SimplifiedMessageSummary(MessageSummary summary, IMailFolder folder, ListStructure attachmentsList) : base(summary.Index)
+        private MessageSummary messageSummary = null;
+        private MimeMessage fullMessage = null;
+
+        public SimplifiedMessageSummary()
         {
-            TextBody = ParseBody(summary, folder);
-            MessageAttachments = attachmentsList;
-            Envelope = summary.Envelope;
-            UniqueId = summary.UniqueId;
+            FullMessage = new MimeMessage();
+        }
+
+        public SimplifiedMessageSummary(MessageSummary summary, IMailFolder folder)
+        {
+            messageSummary = summary;
             Folder = folder;
         }
 
-        private static string ParseBody(IMessageSummary message, IMailFolder folder)
+        protected SimplifiedMessageSummary(MimeMessage message)
+        {
+            FullMessage = message;
+        }
+
+        public MimeMessage FullMessage
+        {
+            get
+            {
+                if (fullMessage == null && messageSummary != null)
+                    fullMessage = Folder.GetMessage(messageSummary.UniqueId);
+                return fullMessage;
+            }
+            set
+            {
+                fullMessage = value;
+            }
+        }
+
+        public List<object> Attachments
+        {
+            get => CreateAttachmentStructuresFromAttachments(FullMessage.Attachments);
+            set
+            {
+                var bodyBuilder = new BodyBuilder();
+                if (bodyBuilder.Attachments != null)
+                {
+                    bodyBuilder.HtmlBody = HtmlBody;
+                    bodyBuilder.TextBody = TextBody;
+                    foreach (var attachment in value)
+                        bodyBuilder.Attachments.Add(attachment.ToString());
+                    FullMessage.Body = bodyBuilder.ToMessageBody();
+                }
+            }
+        }
+
+        private List<object> CreateAttachmentStructuresFromAttachments(IEnumerable<MimeEntity> attachments)
+        {
+            List<object> attachmentsList = new List<object>();
+
+            foreach (var attachment in attachments.Where(x => !string.IsNullOrEmpty(x.ContentDisposition?.FileName)))
+            {
+                AttachmentModel attachmentModel = new AttachmentModel(attachment);
+                AttachmentStructure temp = new AttachmentStructure(attachmentModel);
+                attachmentsList.Add(temp);
+            }
+            return attachmentsList;
+        }
+
+        public string MessageId
+        {
+            get
+            {
+                if (fullMessage != null)
+                    return fullMessage.MessageId;
+                else if (messageSummary != null)
+                    return messageSummary.Envelope.MessageId;
+                return string.Empty;
```

Comment:
Throw exception in such case. But if you split this model into two separate ones then you will just do e.g. `public string MessageId => messageSummary.Envelope.MessageId;`. The code will simplify with each getter taking just one line without any logic inside.